scripts/build/termux_get_repo_files.sh

```diff
@@ -1,64 +1,84 @@
+		case "$TERMUX_REPO_PKG_FORMAT" in
+		"debian")
+			RELEASE_FILE="${TERMUX_COMMON_CACHEDIR}/${TERMUX_REPO_NAME}-${TERMUX_REPO_DISTRIBUTION[$idx]}-Release"
+			repo_base="${TERMUX_REPO_URL[$idx]}/dists/${TERMUX_REPO_DISTRIBUTION[$idx]}"
+			dl_prefix="${TERMUX_REPO_NAME}-${TERMUX_REPO_DISTRIBUTION[$idx]}-${TERMUX_REPO_COMPONENT[$idx]}"
+			RELEASE_FILE_URL="${repo_base}/Release"
+			RELEASE_FILE_SIG_URL="${RELEASE_FILE_URL}.gpg"
+			;;
+		"pacman")
+			RELEASE_FILE="${TERMUX_COMMON_CACHEDIR}-${TERMUX_ARCH}/${TERMUX_REPO_NAME}-json"
+			repo_base="${TERMUX_REPO_URL[$idx]}/${TERMUX_ARCH}"
+			RELEASE_FILE_URL="${repo_base}/${TERMUX_REPO_DISTRIBUTION[$idx]}.json"
+			RELEASE_FILE_SIG_URL="${RELEASE_FILE_URL}.sig"
+			;;
+		esac
 
-		local download_attempts=6
-		while ((download_attempts > 0)); do
-			if [ "$TERMUX_REPO_PKG_FORMAT" = "debian" ]; then
-				if termux_download "${repo_base}/Release" "$RELEASE_FILE" SKIP_CHECKSUM && \
-					termux_download "${repo_base}/Release.gpg" "${RELEASE_FILE}.gpg" SKIP_CHECKSUM && \
-					gpg --verify "${RELEASE_FILE}.gpg" "$RELEASE_FILE"; then
+		(
+			if termux_download "${RELEASE_FILE_URL}" "${RELEASE_FILE}" SKIP_CHECKSUM && \
+				termux_download "${RELEASE_FILE_SIG_URL}" "${RELEASE_FILE}.gpg" SKIP_CHECKSUM && \
+				gpg --verify "${RELEASE_FILE}.gpg" "${RELEASE_FILE}"; then
 
-					local failed=false
-					for arch in all $TERMUX_ARCH; do
-						local PACKAGES_HASH=$(./scripts/get_hash_from_file.py ${RELEASE_FILE} $arch ${TERMUX_REPO_COMPONENT[$idx-1]})
+				if [[ "$TERMUX_REPO_PKG_FORMAT" == "debian" ]]; then
+					for arch in all ${TERMUX_ARCH}; do
+						PACKAGES_HASH=$(./scripts/get_hash_from_file.py ${RELEASE_FILE} ${arch} ${TERMUX_REPO_COMPONENT[$idx]})
 
 						# If packages_hash = "" then the repo probably doesn't contain debs for $arch
-						if [ -n "$PACKAGES_HASH" ]; then
-							if ! termux_download "${repo_base}/${TERMUX_REPO_COMPONENT[$idx-1]}/binary-$arch/Packages" \
-								"${TERMUX_COMMON_CACHEDIR}-$arch/${dl_prefix}-Packages" \
-								$PACKAGES_HASH; then
-								failed=true
-								break
-							fi
+						if [[ -n "$PACKAGES_HASH" ]] && ! termux_download "${repo_base}/${TERMUX_REPO_COMPONENT[$idx]}/binary-$arch/Packages" \
+									"${TERMUX_COMMON_CACHEDIR}-$arch/${dl_prefix}-Packages" \
+									$PACKAGES_HASH; then
+							termux_error_exit "Failed to download package repository metadata. Try to build without -i/-I option."
 						fi
 					done
-
-					if ! $failed; then
-						break
-					fi
-				fi
-			elif [ "$TERMUX_REPO_PKG_FORMAT" = "pacman" ]; then
-				if termux_download "${repo_base}/${TERMUX_REPO_DISTRIBUTION[$idx-1]}.json" "$JSON_FILE" SKIP_CHECKSUM && \
-					termux_download "${repo_base}/${TERMUX_REPO_DISTRIBUTION[$idx-1]}.json.sig" "${JSON_FILE}.sig" SKIP_CHECKSUM && \
-					gpg --verify "${JSON_FILE}.sig" "$JSON_FILE"; then
-
-					break
 				fi
-			fi
-
-			download_attempts=$((download_attempts - 1))
-			if ((download_attempts < 1)); then
+			else
```

Comment:
Can we invert the condition or something to pull this out of the `else` block and either turn it into a guard clause or return inside the `if` block and have this be a fail state that's only reached if that `if` block doesn't run?

Reply:
Done.